decidim-proposals/app/commands/decidim/proposals/publish_proposal.rb

```diff
@@ -20,12 +20,42 @@ def initialize(proposal, current_user)
       #
       # Returns nothing.
       def call
-        return broadcast(:invalid) if @proposal.author != current_user
+        return broadcast(:invalid) if @proposal.author != @current_user
 
         @proposal.update published_at: Time.current
```

Comment:
This should be inside the transaction too

Reply:
Ok, I'll fix it.

Reply:
done! pushed changes